Готова первая проектная работа

nastyaprana opened the pull request:
Отправляю на ревью

nastyaprana committed:
Выполнила первый проект

nastyaprana committed:
Убрала дублирующие свойства

nastyaprana committed:
Убрала дублирующие свойства (2)

nastyaprana committed:
Удалили лишний файл style.css

AliaksandrHrychonak commented on fonts/fonts.css:
```diff
@@ -0,0 +1,43 @@
+@font-face {
+  font-family: EB Garamond;
+  src: local("ebgaramondmedium"), local("EB Garamond-Medium");
```
Надо исправить

В отдельном src для каждого подключения нет необходимости, более точным оформлением, перечислить в одном src подключения через запятую

AliaksandrHrychonak commented on index.html:
```diff
@@ -0,0 +1,214 @@
+    <link rel="stylesheet" href="./styles/style.css" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <title>С чистого листа</title>
+  </head>
```
Верно оформлен `head`

AliaksandrHrychonak commented on index.html:
```diff
@@ -0,0 +1,214 @@
+      </a>
+      <nav class="header__menu">
+        <ul class="header__menu-list">
+          <li class="header__menu-item_active">
```
Надо исправить

Необходимо добавить общую сущность `header__menu-item` на случай если активный элемент сменится

AliaksandrHrychonak commented on index.html:
```diff
@@ -0,0 +1,214 @@
+        </ul>
+      </nav>
+    </header>
+    <main>
```
Отлично

Контентное содержимое обёрнуто тегом `main`

AliaksandrHrychonak commented on index.html:
```diff
@@ -0,0 +1,214 @@
+          состояние потока. Причин, почему первые усилия даются так сложно,
+          может быть очень много. Вот основные:
+        </p>
+        <ul class="content-section__list">
```
Отличное решение использовать список

AliaksandrHrychonak commented on index.html:
```diff
@@ -0,0 +1,214 @@
+      <div class="footer__copy">© VЁRSTKA</div>
+      <div>
+        <ul class="footer__links">
+          <li class="footer__links-vk">ВК</li>
```
Надо исправить

Каждый текст нужно сделать ссылкой.
Класс для стилизации нужен одинаковый, так как стили элементов однотипны

AliaksandrHrychonak commented on index.html:
```diff
@@ -0,0 +1,214 @@
+      <div>
+        <ul class="footer__links">
+          <li class="footer__links-vk">ВК</li>
+          <li class="footer__links-youtube">Ютуб</li>
```
Надо исправить

Каждый текст нужно сделать ссылкой.
Класс для стилизации нужен одинаковый, так как стили элементов однотипны

AliaksandrHrychonak commented on index.html:
```diff
@@ -0,0 +1,214 @@
+      </section>
+    </main>
+    <footer class="content-section content-section_theme_dark footer">
+      <div class="footer__copy">© VЁRSTKA</div>
```
Надо исправить

`div` не подходящее решение для разметки текста. Используйте тег `<p>`

AliaksandrHrychonak commented on styles/style.css:
```diff
@@ -0,0 +1,272 @@
+
+header {
+  width: 360px;
+  height: 132px;
```
Надо исправить

Жёсткую фиксацию высоту необходимо удалить.

Не рекомендуется задавать фиксированную высоту блокам, так как это может привести к проблемам с отображением при изменении количества контента. При добавлении контента он может выйти за пределы блока, а при удалении может образоваться лишнее пустое пространство.

Высота блока должна определяться его содержимым автоматически. Это обеспечит корректное отображение независимо от количества и размера контента внутри блока.

Такой подход делает отображение более гибким и устойчивым к изменениям в будущем, когда может потребоваться добавить или удалить контент.

AliaksandrHrychonak commented on styles/style.css:
```diff
@@ -0,0 +1,272 @@
+}
+
+.header__menu {
+  height: 84px;
```
Надо исправить

Жёсткую фиксацию высоту необходимо удалить.

Не рекомендуется задавать фиксированную высоту блокам, так как это может привести к проблемам с отображением при изменении количества контента. При добавлении контента он может выйти за пределы блока, а при удалении может образоваться лишнее пустое пространство.

Высота блока должна определяться его содержимым автоматически. Это обеспечит корректное отображение независимо от количества и размера контента внутри блока.

Такой подход делает отображение более гибким и устойчивым к изменениям в будущем, когда может потребоваться добавить или удалить контент.

AliaksandrHrychonak commented on styles/style.css:
```diff
@@ -0,0 +1,272 @@
+  line-height: 29.7px;
+}
+
+.card__image {
```
Надо исправить

Чтобы сохранить пропорциональное скругление углов независимо от размера изображения, используйте процентное значение: `border-radius: 50%`. Это значение всегда будет вычисляться относительно текущих размеров элемента.

AliaksandrHrychonak commented on styles/style.css:
```diff
@@ -0,0 +1,272 @@
+}
+
+.intro-title {
+  display: inline-block;
```
Надо исправить

Использование display: inline/inline-block для построения сетки является устаревшим подходом, который создает множество технических проблем. При работе с inline-block элементами возникают нежелательные пробелы между компонентами, которые приходится обходить хаками с отрицательными марджинами или манипуляциями с font-size. Вертикальное выравнивание становится настоящей головной болью из-за особенностей работы baseline. При адаптивной верстке inline-block элементы ведут себя непредсказуемо, часто нарушая задуманный макет.

Современные технологии вроде Flexbox и Grid предоставляют гораздо более мощные и интуитивно понятные инструменты. Flexbox позволяет легко управлять распределением пространства между элементами, их выравниванием и порядком отображения. Достаточно задать display: flex родительскому элементу, и мы получаем доступ к богатому набору свойств вроде justify-content, align-items и flex-grow, которые решают типичные задачи верстки в несколько строк кода.

Grid идет еще дальше, предоставляя полноценную двумерную систему компоновки. С помощью grid-template-columns и grid-template-rows можно создавать сложные сетки с точным контролем размеров и отступов. Grid Areas позволяют давать осмысленные имена участкам сетки и оперировать ими при построении адаптивного дизайна. Автоматическое размещение элементов в Grid избавляет от необходимости явно прописывать правила для каждого брейкпоинта.

Оба этих подхода имеют отличную поддержку браузерами и являются стандартом современной веб-разработки. Они обеспечивают чистый, поддерживаемый код без костылей и хаков, необходимых при работе с inline-block.

AliaksandrHrychonak commented on styles/style.css:
```diff
@@ -0,0 +1,272 @@
+  text-decoration: underline;
+  padding-bottom: 12px;
+}
+.header__menu-item:last-child {
```
Можно лучше

Удобно использовать gap для создания разрывов 

https://doka.guide/css/gap/

AliaksandrHrychonak reviewed (commented):
Необходимо развернуть общий комментарий ⬇️

Добрый день. Работа выполнена на хорошем уровне! 

Отлично:

* аккуратно оформленный код
* все комментарии "Отлично"

Надо исправить:

* Все комментарии с пометкой "надо исправить"
- Во всем проекте для всех текстовых элементов и обёрток используется `font-family: Шрифт`, это является не совсем корректным решением. В том случае, если по какой-либо причине шрифты не загрузятся или будет использоваться браузер, не поддерживающий текущий формат шрифта - следует предусмотреть такой исход и добавить системные шрифты. **Необходимо подправить этот момент во всём коде**, нужно добавлять альтернативные шрифты, если `Шрифт` не получилось найти (загрузить). И обязательно в самом конце указать `sans-serif (без засечек)`, иначе браузер применит какой-нибудь стандартный шрифт, который найдет.
* В проекте есть нарушения в отображении по макету 
    - https://disk.yandex.com/i/mLSv3IGVCtrOTg
    * Для более точного контроля отображения вёрстки можно воспользоваться плагином `PerfectPixel` для `Chrome`, можно в фигме выделить весь макет, экспортировать его картинкой и наложить на вёрстку в браузере. Плагин позволяет наложить изображение на вёрстку в браузере, тем самым можно слегкостью подправить недочёты по отображению. 


Можно лучше:

* Все комментарии "Можно лучше" в коде.


Необходимо исправить все "надо исправить". Исправление "можно лучше" остаются на ваше усмотрение.

nastyaprana committed:
Внесла правки после ревью

Поправила нарушения в отображении по макету (использовала PerfectPixel), а также внесла правки согласно комментариям:
1. Добавила sans-serif в конце для шрифтов
2. Убрала дубли src в подключении шрифтов
3. В шапке добавила общий header__menu-item чтобы был у всех li (если сменится активный элемент)
4. В футере текст сделала ссылками + классы для стилизации сделала одинаковыми
5. Для текста использовала p вместо ошибочного div
6. Убрала в 2х местах жесткую фиксацию по высоте
7. Для изображений добавила border-radius: 50% - для сохранения пропорционального скругления углов
8. Заменила display: inline/inline-block на display: flex
9. В шапке и футере использовала gap для создания разрывов

nastyaprana committed:
Небольшие правки после тестов

AliaksandrHrychonak commented on styles/style.css:
```diff
@@ -0,0 +1,277 @@
+}
+
+.footer__links {
+  font-family: Inter;
```
Надо исправить

Используется `font-family: Шрифт`, это является не совсем корректным решением. В том случае, если по какой-либо причине шрифты не загрузятся или будет использоваться браузер, не поддерживающий текущий формат шрифта - следует предусмотреть такой исход и добавить системные шрифты. **Необходимо подправить этот момент во всём коде**, нужно добавлять альтернативные шрифты, если `Шрифт` не получилось найти (загрузить). И обязательно в самом конце указать `sans-serif (без засечек)`, иначе браузер применит какой-нибудь стандартный шрифт, который найдет.

AliaksandrHrychonak reviewed (commented):
Добрый день, отличные исправления, осталось исправить все комментарии в коде
